lib/galaxy/tools/actions/metadata.py

```diff
@@ -66,7 +66,7 @@ def execute_via_app(self, tool, app, session_id, history_id, user=None,
         except AttributeError:
             job.tool_version = "1.0.1"
         job.dynamic_tool = tool.dynamic_tool
-        job.state = job.states.WAITING  # we need to set job state to something other than NEW, or else when tracking jobs in db it will be picked up before we have added input / output parameters
+        job.state = job.states.NOT_READY    # we need to set job state to something other than NEW, or else when tracking jobs in db it will be picked up before we have added input / output parameters
```

Comment:
```suggestion
        job.state = job.states.NOT_READY  # we need to set job state to something other than NEW, or else when tracking jobs in db it will be picked up before we have added input / output parameters
```